bgzip.c

```diff
@@ -312,8 +296,17 @@ int main(int argc, char **argv)
             }
             else {
                 const int wrflags = O_WRONLY | O_CREAT | O_TRUNC;
+                char *name = argv[optind];
```

Comment:
Is there a danger here that if you tried to `bgzip -d -f something.bam.csi` then you might overwrite `something.bam` unintentionally?  Or given that it's necessary to use the `-f` option for this to happen, do we not worry as it was explicitly requested?

Reply:
Good point, I guess that is a new danger. I guess the possible answers are

1. Whitelist known compression extensions, namely `.gz` and `.bgz` (and …?)

2. Blacklist known index extensions, e.g., `.bai`, `.csi`, etc

3. Accept any extensions like this code does, and depend on the `-f` failsafe; if they say yes to that, they really do mean it.

4. Status quo, which is basically "only `.gz` is whitelisted"

Hmmm… option 3 and change to one of the others if problems ensue?
I guess (4) suggests that (1) is the minimal change…

Reply:
Sticking to `.gz` and `.bgz` would be safest.  Or we could make it politely refuse if the output file exists even if you've used `-f` _unless_ one of the known-safe extensions is used (and allow something like `-f -f` for "I really really want to do this").

Reply:
PR updated to do (1) essentially — asks you both “are you sure about this funny extension” and “are you sure about overwriting” as appropriate, and both questions can be suppressed with enough `-f -f`.